apps/site/assets/ts/hooks/useSchedules.ts

```diff
@@ -36,14 +38,21 @@ const fetchData = async (url: string): Promise<ScheduleData[]> =>
 const useSchedulesByStop = (
   stopId: string
 ): FetchState<ScheduleWithTimestamp[]> => {
+  const [updateDate, setUpdateDate] = useState(Date.now());
+  useInterval(() => {
+    setUpdateDate(Date.now());
+  }, 15000);
+
   const { data, error } = useSWR<ScheduleData[]>(
     `/api/stops/${stopId}/schedules?last_stop_departures=false&future_departures=true`,
     fetchData
   );
   if (error) {
     return { status: FetchStatus.Error };
   }
-  const parsedData = data?.map(d => parse(d));
+  const parsedData = data
+    ?.map(d => parse(d))
+    .filter(d => d.time.getTime() >= updateDate);
```

Comment:
So this is filtering all the schedules that are before the `updateDate`?

Reply:
Yeah, excluding all schedules before the ever-changing `updateDate`.